illumina/GetSampleInfoForSampleSheet.py

```
@@ -21,16 +21,35 @@ def getTWISTinfo(planID, DOM, Name, sLimsID, sUDFs):
     
     SampleID = sLimsID + "_" + planID
 
-    DiagnosisDict = { "Hematologisk neoplasi" : "Hem" ,
+    if "GMSMyeloid" in sUDFs["Analysis"] or "Ovarial" in sUDFs["Analysis"] or "GMSLymphoma" in sUDFs["Analysis"] or "GMSSolidTumor" in sUDFs["Analysis"]:
+        #Get Tumor/Normal
+
+        Cellularity = ""
+
+        if "GMSSolidTumor" in sUDFs["Analysis"]:
+            try:
+                Cellularity = str(sUDFs["Cellularity (%)"])
+            except KeyError:
+                print("Cellularity (%) is missing.")
+
+            DiagnosisDict = { u"Hjärntumör (CNS)" : "cns" ,
+                      u"Lungcancer/sköldkörtelcancer" : "lungthyroid",
+                      u"Kolorektalcancer" : "colon" , 
+                      u"Bröstcancer" : "breast", 
+                      u"Äggstockscancer" : "ovarian",
+                      u"Gastrointestinal stromal tumor (GIST)" : "gist",
+                      u"Malignt melanom" : "melanoma",
+                      u"Bröstcancerpilot" : "BP",
+                      u"Solid tumör, övriga neoplasier" : "solid"}
+        else:
+            DiagnosisDict = { "Hematologisk neoplasi" : "Hem" ,
                       "KLL" : "KLL",
                       "MPN" : "MPN" ,
                       "Annat" : "other",
                       "Hematologisk neoplasi efter transplantation" : "AlloSCT",
 		      "Genpanelriktad" : "Riktad",
               "Hematologisk neoplasi, snabbsvar" : "Hem-Snabb" }
-
-    if "GMSMyeloid" in sUDFs["Analysis"] or "Ovarial" in sUDFs["Analysis"] or "GMSLymphoma" in sUDFs["Analysis"] or "GMSSolidTumor" in sUDFs["Analysis"]:
-        #Get Tumor/Normal
+            
 
```

Comment about row 76. It looks like if a user has left the UDF/Diagnosis field empty it will be other in the description field, is that the case or am I missing something?
This should not be possible for GMSSolidTumor. If no value is give, an error should be thrown, the script fail and the user should be told which sample is missing a value for Diagnosis.

---

Good point, will fix.